src/compiler/checker.ts

```diff
@@ -10765,11 +10765,15 @@ namespace ts {
             let typeType = valueType;
             if (symbol.valueDeclaration) {
                 const decl = getRootDeclaration(symbol.valueDeclaration);
-                const isRequireAlias = isVariableDeclaration(decl)
-                    && decl.initializer
-                    && isCallExpression(decl.initializer)
-                    && isRequireCall(decl.initializer, /*requireStringLiteralLikeArgument*/ true)
-                    && valueType.symbol;
+                let isRequireAlias = false;
+                if (isVariableDeclaration(decl) && decl.initializer) {
+                    let expr = decl.initializer;
+                    // skip past entity names, eg `require("x").a.b.c`
+                    while (isPropertyAccessExpression(expr)) {

```

Comment:
What about optional chains?

Reply:
Those work -- I added a test. I guess they use the same AST as normal property access expressions?